### PR TITLE
Add doppler_project_role resource

### PR DESCRIPTION
**Pull request by nmanoogian**
Closes ENG-7582

### CODE REVIEW

File: docs/resources/project_role.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `name` (String) The name of the Doppler project role
+- `permissions` (List of String) A list of Doppler project permissions
```
Comment (watsonian) [inline docs/resources/project_role.md:27]: Can we link to the [API docs](https://docs.doppler.com/reference/project_roles-create) that contain the list of valid permission values here? Would be a nice QoL thing for anyone looking at the TF docs.

File: doppler/api.go [hunk 2]
```diff
@@ -337,9 +337,71 @@ func (client APIClient) DeleteProject(ctx context.Context, name string) error {
...
+	_, err := client.PerformRequestWithRetry(ctx, "DELETE", fmt.Sprintf("/v3/projects/roles/role/%s", url.PathEscape(identifier)), []QueryParam{}, nil)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Project Members
 func (client APIClient) CreateProjectMember(ctx context.Context, project string, memberType string, memberSlug string, role string, environments []string) (*ProjectMember, error) {
+
```
Comment (watsonian) [inline doppler/api.go:404]: Stray newline.

**Review by watsonian — APPROVED** [review]
Looks good! Left a couple notes, but no blockers from my perspective. Note that I didn't test this locally though.

File: docs/resources/project_role.md (hunk 1 above)
Comment (nmanoogian) [reply docs/resources/project_role.md:27]: Good call 👍

**Comment by nmanoogian** [comment]
[Force push](https://github.com/DopplerHQ/terraform-provider-doppler/compare/f79f9765819f0cbf6d686ced03f77e9922d9f4d1..704208fb4f0b12c90408d5e83221f460002e9e9f) addresses @watsonian's feedback (and rebuilding the docs also showed that I forgot to run `make tfdocs` after switching from a list to a set)

**Pull request merged**